### PR TITLE
update json-schema draft

### LINKED ISSUE #4784: json-schema draft-05 (404). Update to draft-07
### Describe the bug

https://json-schema.org/draft-05/schema# is a 404. Updating to https://json-schema.org/draft-07/schema# will help those who rely on this schema being available.

### Fabric8 Kubernetes Client version

SNAPSHOT

### Steps to reproduce

1. Do anything that relies on "http://json-schema.org/draft-05/schema#" not being a 404
2. Get error.

### Expected behavior

the designated json-schema is available to use. 

### Runtime

OpenShift

### Kubernetes API Server version

1.25.3@latest

### Environment

macOS

### Fabric8 Kubernetes Client Logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AustinRedenbaugh**
http://json-schema.org/draft-05/schema# is a 404... updating to latest version of schema.
Fixes #4784

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
Thanks for your PR :+1: 

JsonSchema Draft 05 seems to be outdated[0]. Do you think this migration can have some backward incompatible[1] changes from the KubernetesClient user perspective?

[0] https://json-schema.org/draft-06/json-schema-release-notes.html#q-what-happened-to-draft-05
[1] https://json-schema.org/draft-06/json-schema-release-notes.html#q-what-are-the-changes-between-draft-04-and-draft-06

File: CHANGELOG.md [hunk 1]
```diff
@@ -950,7 +950,7 @@ Please see the [migration guide](doc/MIGRATION-v6.md)
 ### 5.0.0-alpha-3 (2020-12-10)
 
 #### Bugs
-* Fix #2519: Generated schemas contains a valid meta-schema URI reference (`http://json-schema.org/draft-05/schema#`)
+* Fix #2519: Generated schemas contains a valid meta-schema URI reference (`http://json-schema.org/draft-07/schema#`)
```
Comment (manusa) [inline CHANGELOG.md:953]: Should not be changed

+ Should contain an entry for 6.5-SNAPSHOT with the change

File: doc/MIGRATION-v5.md [hunk 2]
```diff
@@ -268,13 +268,13 @@ We have added some minor improvements in shared informer APIs. Changes are as fo
 `kube-validation-schema.json` files are no longer included in our model packaged. The equivalent file
 is now named `validation-schema.json`.
 
-The [schema][json-schema] URL has been updated to the latest compatible [version][json-schema-draft-05]
-(`http://json-schema.org/draft-05/schema#`)
+The [schema][json-schema] URL has been updated to the latest compatible [version][json-schema-draft-07]
```
Comment (manusa) [inline doc/MIGRATION-v5.md:271]: Should not be changed

**Comment by manusa** [comment]
I think the latest draft is `2020-12`: https://json-schema.org/draft/2020-12/schema

**Comment by pdodds** [comment]
I think @AustinRedenbaugh went the closest next one, we have been running the code with that version in place in a production env for a few months.

**Comment by manusa** [comment]
> I think @AustinRedenbaugh went the closest next one, we have been running the code with that version in place in a production env for a few months.

Cool, so if you confirm that this works, we can bump to 07 for the time being. We can merge once the minor comments are addressed and we update the description.

File: CHANGELOG.md (hunk 1 above)
Comment (AustinRedenbaugh) [reply CHANGELOG.md]: resolved.

File: doc/MIGRATION-v5.md (hunk 2 above)
Comment (AustinRedenbaugh) [reply doc/MIGRATION-v5.md]: resolved.

**Comment by AustinRedenbaugh** [comment]
What are the pending items on this to get this through?

..and thank you.
Austin

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**